lib/rules/media-type-no-deprecated/index.mjs

```diff
@@ -0,0 +1,72 @@
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { atRuleRegexes } from '../../utils/regexes.mjs';
+import { deprecatedMediaTypesNames } from '../../reference/mediaTypes.mjs';
+import { isMediaQueryWithType } from '@csstools/media-query-list-parser';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import parseMediaQuery from '../../utils/parseMediaQuery.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'media-type-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (mediaType) => `Unexpected deprecated media type "${mediaType}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-type-no-deprecated',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreMediaType: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkAtRules(atRuleRegexes.mediaName, (atRule) => {
+			const mediaQueryList = parseMediaQuery(atRule);
+
+			mediaQueryList.forEach((mediaQuery) => {
```

Comment:
@immitsu Can you add a regexp check before parsing?
We want to avoid parsing when we already know it isn't needed.

So when an atrule doesn't contain any of `tty`, `tv`, ... we can return early, avoiding more costly compute.